studies/calgames_kinematics/src/main/java/org/team100/Kinematics.java

```diff
@@ -1,24 +1,101 @@
+
+        System.out.println("\nOUTPUTS:");
+        System.out.println("armXcomp: " + armXcomp);
+        System.out.println("armYcomp: " + armYcomp);
+        System.out.println("manipulatorXcomp: " + manipulatorXcomp);
+        System.out.println("manipulatorYcomp: " + manipulatorYcomp);
+        System.out.println("X Point: " + (xComp));
+        System.out.println("Y Point: " + yComp);
+        System.out.println("Manipulator Rotation: " + (manipulatorRotation));
+
+        return new Pose2d(xComp, yComp, new Rotation2d(manipulatorRotation));
     }
 
     public Config inverse(Pose2d pose) {
         // inverse kinematics
-        return new Config(0, 0, 0);
+
+        System.out.println("\nTESTING INVERSE KINEMATICS:\n" );
+
+        System.out.println("PRESETS:" );
+        System.out.println("Arm Length: " + m_armLength);
+        System.out.println("Manipulator Length: " + m_manipulatorLength);
+        System.out.println("Goal X: " + pose.getX());
+        System.out.println("Goal Y: " + pose.getY());
+        System.out.println("Goal Rotation: " + pose.getRotation().getDegrees());
+
+        //1. find the elbow position from the manipulator length and the goal 2dpose. ('p' in my notes) - KYM (CORRECT - 9/13/2025)
+        m_elbowPointY = m_manipulatorLength*Math.sin(pose.getRotation().getRadians()) + pose.getY(); // vert side of triangle with hyp facing right, add key point height
+        m_elbowPointX = pose.getX()-(m_manipulatorLength*Math.cos(pose.getRotation().getRadians())); //bottom of triangle with hyp facing right, subtract from key point
+
+        
+        //2. find the third leg of the triange formed by elbowPointX, hyp = armLength, and subtract that from the height 
+        //of elbowpointY to find the actual shoulder joint height - KYM
+        m_shoulderHeight = m_elbowPointY-(Math.sqrt((m_armLength*m_armLength)-(m_elbowPointX*m_elbowPointX)));
+
+
+        //find last side of triangle formed by both arms, vertex's being shoulderPt, goalPt, and elbowPt
+        m_zLength = Math.sqrt((pose.getX()*(pose.getX()))+((pose.getY()-m_shoulderHeight)*(pose.getY()-m_shoulderHeight)));
+        
+        //law of cosines to find the angle INSIDE the triangle that corresponds to the shoulder joint. second part is angle from 0 degrees to angle of elevation. gnarly
+        //this part needs to be edited for going negative angles with arm?
+        m_shoulderAngle = (Math.acos((m_armLength*m_armLength + m_zLength*m_zLength - m_manipulatorLength*m_manipulatorLength) / (2 * m_armLength * m_zLength)))+((Math.asin((pose.getY()-m_shoulderHeight)/m_zLength)));
+
+        //first part is law of cosidnes but for the elbow instead. 
+        //need to think about how this will work with negative angles?
+        m_elbowAngle = ((Math.acos((m_manipulatorLength*m_manipulatorLength + m_armLength*m_armLength - m_zLength*m_zLength) / (2 * m_manipulatorLength * m_armLength))));
+        
+        System.out.println("\nOUTPUTS:");
```

Comment:
when you find yourself writing print statements like this, it's a hint that you should break the method into smaller methods, so you can test their outputs with unit tests.